lib/mysqlDaoQueryHelper.js

```diff
@@ -79,6 +79,39 @@ class MysqlDaoQueryHelper {
     }
 
     /**
+     * @param {Array.<object>} insertArgsArray
+     * @param {object|Array} onDupUpdateArgs If an object, update columns with values defined by the object keys.  If
+     * an array, update the listed columns with the insert value.
+     * @returns {string}
+     */
+    upsertBulk(insertArgsArray, onDupUpdateArgs) {
+        const cleanValues = [];
+        insertArgsArray.forEach(insertArgsArray => {
+            cleanValues.push(this._cleanAndMapValues(insertArgsArray));
+        });
+
+        const upsert =
+            Squel.useFlavour('mysql').insert().into(this._tableName)
+                .setFieldsRows(cleanValues, {dontQuote: true});
+
+        if (_.isArray(onDupUpdateArgs)) {
+            onDupUpdateArgs.forEach(propName => {
+                const snakeCaseProp = _.snakeCase(propName);
+                upsert.onDupUpdate(snakeCaseProp, "VALUES(" + snakeCaseProp + ")", {dontQuote: true});
+            });
+        } else {
+            const cleanOnDupUpdateValues = this._cleanAndMapValues(onDupUpdateArgs);
+            Object.keys(cleanOnDupUpdateValues).forEach(property => {
```

Comment:
You have lodash in this class, so you can do `_.forEach(cleanOnDupUpdateValues, (val, key) => {});` if you like